dev-docs/best-practices/microservices.md

```diff
@@ -0,0 +1,111 @@
+# Building Microservices
+
+All Taskcluster services are implemented in subdirectories of the the `services/` directory in this respository.
+
+The Taskcluster team shares responsibility for all services, although people may "specialize" in specific services and have additional expertise there.
+This shared responsibility is easier for everyone if the implementations are similar, avoiding surprises when moving from one service to another.
+This document aims to collect the practices and standards we've agreed on.
+
+These conventions are strongly encouraged for new services and contributions updating existing services to follow them are always welcome.
+When we have a good reason to not follow the convention for a specific service, we document why.
+
+## Independence
+
+Although they ar developed in the same repository, each service must operate as an independent entity, using only the public interfaces of other services.
+Package imports between services are strictly forbidden.
+Where this seems like a good idea, prefer to create a new [library](libraries) instead.
+
+## Package Mechanics
+
+A service has a `serviceName` and a `projectName`, the latter often having a `taskclsuter-` prefix.
```

Comment:
We should have a meta-test that catches tskclsuter as well.

Reply:
I have myriad ways of misspelling taskcluster :)